perf_test/CMakeLists.txt

```diff
@@ -50,5 +50,95 @@ if (KokkosKernels_ENABLE_PERFTESTS)
     ADD_COMPONENT_SUBDIRECTORY(blas)
     ADD_SUBDIRECTORY(performance)
     #ADD_SUBDIRECTORY(common)
+
+    IF (KOKKOS_HAS_TRILINOS)
+      message(FATAL_ERROR "Benchmarks are not supported when building as part of Trilinos")
+    ENDIF()
+
+    find_package(benchmark QUIET)
+    IF(benchmark_FOUND)
+      MESSAGE(STATUS "Using google benchmark found in ${benchmark_DIR}")
+    ELSE()
+      message(STATUS "No installed google benchmark found, fetching from GitHub")
+      include(FetchContent)
+      SET(BENCHMARK_ENABLE_TESTING OFF)
+
+      list(APPEND CMAKE_MESSAGE_INDENT "    ")
+      FetchContent_Declare(
+        googlebenchmark
+        URL https://github.com/google/benchmark/archive/refs/tags/v1.6.2.tar.gz
+        URL_HASH MD5=14d14849e075af116143a161bc3b927b
```

Comment:
we need to add a comment here to clarify why this specific hash was chosen

Reply:
:+1:  d953f8f75e029f5929d4c02c3c3c8319a413cea3